src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs

```diff
@@ -0,0 +1,177 @@
+            var wallet = this.nodes[PosStaker].FullNode.WalletManager().GetWalletByName(PosWallet);
+            this.posReceiverPrivateKey = wallet.GetExtendedPrivateKeyForAddress(PosWalletPassword, this.posReceiverAddress).PrivateKey;
+
+            this.nodes[PosStaker].SetDummyMinerSecret(new BitcoinSecret(this.posReceiverPrivateKey, this.nodes[PosStaker].FullNode.Network));
+            this.nodeGroupBuilder.WithConnections().Connect(PowMiner, PosStaker);
+        }
+
+        private void create_tx_to_send_million_coins_from_pow_wallet_to_pos_node_wallet()
+        {
+            var context = SharedSteps.CreateTransactionBuildContext(
+                PowWallet,
+                PowWalletAccount,
+                PowWalletPassword,
+                this.posReceiverAddress.ScriptPubKey,
+                Money.COIN * 1000000,
+                FeeType.Medium,
+                101);
+
+            context.OverrideFeeRate = new FeeRate(Money.Satoshis(20000));
+
+            var sendTransaction = CreateSendTransaction();
+            this.nodes[PowMiner].FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(sendTransaction.ToHex()));
+        }
+
+        private Transaction CreateSendTransaction()
+        {
+            var unspent = this.nodes[PowMiner].FullNode.WalletManager().GetSpendableTransactionsInWallet(PowWallet);
+            var coins = new List<Coin>();
+
+            var blockTimestamp = unspent.OrderBy(u => u.Transaction.CreationTime).Select(ts => ts.Transaction.CreationTime).First();
+            var transaction = new Transaction
+            {
+                Time = (uint)blockTimestamp.ToUnixTimeSeconds()
+            };
+
+            foreach (var item in unspent.OrderByDescending(a => a.Transaction.Amount))
+            {
+                coins.Add(new Coin(item.Transaction.Id, (uint)item.Transaction.Index, item.Transaction.Amount, item.Transaction.ScriptPubKey));
+            }
+
+            var coin = coins.First();
+            var txIn = transaction.AddInput(new TxIn(coin.Outpoint, this.powSenderAddress.ScriptPubKey));
+            transaction.AddOutput(new TxOut(new Money(9699999999995400), this.powSenderAddress.ScriptPubKey));
+            transaction.AddOutput(new TxOut(new Money(100000000000000), this.posReceiverAddress.ScriptPubKey));
+            transaction.Sign(this.powSenderPrivateKey, new[] { coin });
+
+            return transaction;
+        }
+
+        private void pow_wallet_broadcasts_tx_of_million_coins_and_pos_wallet_receives()
+        {
+            // Wait for the coins to arrive
+            TestHelper.WaitLoop(() => this.nodes[PosStaker].CreateRPCClient().GetRawMempool().Length > 0);
+            TestHelper.WaitLoop(() => this.nodes[PosStaker].FullNode.WalletManager().GetSpendableTransactionsInWallet(PosWallet).Any());
```

Comment:
@ferdeen had some extension methods in his PR which might clean these lines up a lot, but I don't think that is quite merged in yet